v1.4.0 doesn't work anymore

ended with exception that PlexusContainerImpl doesn't have any default constructor (what seems to be true)

```
java.lang.RuntimeException: io.quarkus.bootstrap.BootstrapException: Failed to create the application model for ch.medidata.mdc:mdc-logging-jaxrs::jar:local-SNAPSHOTnull
        at io.quarkus.test.junit.QuarkusTestExtension.throwBootFailureException(QuarkusTestExtension.java:638)
        at io.quarkus.test.junit.QuarkusTestExtension.interceptTestClassConstructor(QuarkusTestExtension.java:722)
        at java.base/java.util.Optional.orElseGet(Optional.java:364)
        at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
        at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
Caused by: io.quarkus.bootstrap.BootstrapException: Failed to create the application model for ch.medidata.mdc:mdc-logging-jaxrs::jar:local-SNAPSHOTnull
        at io.quarkus.bootstrap.BootstrapAppModelFactory.resolveAppModel(BootstrapAppModelFactory.java:297)
        at io.quarkus.bootstrap.app.QuarkusBootstrap.bootstrap(QuarkusBootstrap.java:133)
        at io.quarkus.test.junit.AbstractJvmQuarkusTestExtension.createAugmentor(AbstractJvmQuarkusTestExtension.java:189)
        at io.quarkus.test.junit.QuarkusTestExtension.doJavaStart(QuarkusTestExtension.java:219)
        at io.quarkus.test.junit.QuarkusTestExtension.ensureStarted(QuarkusTestExtension.java:605)
        at io.quarkus.test.junit.QuarkusTestExtension.beforeAll(QuarkusTestExtension.java:655)
        ... 1 more
Caused by: java.lang.RuntimeException: Failed to create application model resolver for D:\git\mdc\libs\mdc-logging\jaxrs-runtime
        at io.quarkus.bootstrap.BootstrapAppModelFactory.getAppModelResolver(BootstrapAppModelFactory.java:169)
        at io.quarkus.bootstrap.BootstrapAppModelFactory.resolveAppModel(BootstrapAppModelFactory.java:282)
        ... 6 more
Caused by: java.lang.RuntimeException: No valid constructor found on class io.smallrye.beanbag.maven.PlexusContainerImpl
        at io.smallrye.beanbag.sisu.Sisu.findConstructor(Sisu.java:893)
        at io.smallrye.beanbag.sisu.Sisu.addClass(Sisu.java:568)
        at io.smallrye.beanbag.maven.MavenFactory.<init>(MavenFactory.java:65)
        at io.smallrye.beanbag.maven.MavenFactory.create(MavenFactory.java:92)
        at io.smallrye.beanbag.maven.MavenFactory.create(MavenFactory.java:107)
        at io.smallrye.beanbag.maven.MavenFactory.create(MavenFactory.java:120)
        at io.quarkus.bootstrap.resolver.maven.BootstrapMavenContext.configureMavenFactory(BootstrapMavenContext.java:876)
        at io.quarkus.bootstrap.resolver.maven.BootstrapMavenContext.initRepoSystemAndManager(BootstrapMavenContext.java:863)
        at io.quarkus.bootstrap.resolver.maven.BootstrapMavenContext.getRepositorySystem(BootstrapMavenContext.java:279)
        at io.quarkus.bootstrap.resolver.maven.MavenArtifactResolver.<init>(MavenArtifactResolver.java:114)
        at io.quarkus.bootstrap.BootstrapAppModelFactory.getAppModelResolver(BootstrapAppModelFactory.java:163)
        ... 7 more
```

--- a/maven/src/main/java/io/smallrye/beanbag/maven/MavenFactory.java
+++ b/maven/src/main/java/io/smallrye/beanbag/maven/MavenFactory.java
@@ -62,7 +62,11 @@ public final class MavenFactory {
                 .setSupplier(Scope::getContainer)
                 .build();
         // add our simple plexus container
-        sisu.addClass(PlexusContainerImpl.class, dependencyFilter);
+        builder.addBean(PlexusContainerImpl.class)
+                .setPriority(-100)
+                .setSingleton(true)
+                .setSupplier(scope -> new PlexusContainerImpl(scope.getContainer()))
+                .build();
         for (ClassLoader classLoader : classLoaders) {
             sisu.addClassLoader(classLoader, dependencyFilter);
         }

--- a/maven/src/main/java/io/smallrye/beanbag/maven/PlexusContainerImpl.java
+++ b/maven/src/main/java/io/smallrye/beanbag/maven/PlexusContainerImpl.java
@@ -4,11 +4,6 @@ import java.util.Collections;
 import java.util.List;
 import java.util.Map;
 
-import javax.annotation.Priority;
-import javax.inject.Inject;
-import javax.inject.Named;
-import javax.inject.Singleton;
-
 import org.codehaus.plexus.PlexusContainer;
 import org.codehaus.plexus.classworlds.realm.ClassRealm;
 import org.codehaus.plexus.component.composition.CycleDetectedInComponentGraphException;
@@ -20,14 +15,10 @@ import org.codehaus.plexus.context.Context;
 import io.smallrye.beanbag.BeanBag;
 import io.smallrye.beanbag.DependencyFilter;
 
-@Named()
-@Priority(-100)
-@Singleton
 class PlexusContainerImpl implements PlexusContainer {
     private final BeanBag bb;
 
-    @Inject
-    public PlexusContainerImpl(final BeanBag bb) {
+    PlexusContainerImpl(final BeanBag bb) {
         this.bb = bb;
     }
 

--- a/maven/src/test/java/io/smallrye/beanbag/maven/MavenFactoryTestCase.java
+++ b/maven/src/test/java/io/smallrye/beanbag/maven/MavenFactoryTestCase.java
@@ -60,6 +60,13 @@ public final class MavenFactoryTestCase {
     }
 
     @Test
+    public void testPlexusContainerImpl() throws ComponentLookupException {
+        final MavenFactory mavenFactory = MavenFactory.create(MavenFactory.class.getClassLoader());
+        PlexusContainer pc = mavenFactory.getContainer().requireBean(PlexusContainerImpl.class);
+        pc.lookup(SecDispatcher.class);
+    }
+
+    @Test
     public void testWagonThings() {
         final MavenFactory mavenFactory = MavenFactory.create(MavenFactory.class.getClassLoader());
         mavenFactory.getContainer().requireBean(Wagon.class);

